src/values.js

```diff
@@ -360,7 +360,26 @@ function Query(value) {
 util.inherits(Query, Value)
 
 wrapToString(Query, function() {
-  return 'Query(' + Expr.toString(this.value) + ')'
+  const metadata = {
+    api_version: +this.value.api_version,
+  }
+
+  const notEmptyMetadata = Object.keys(metadata).some(key => metadata[key])
+
+  let stringified
+  if (notEmptyMetadata) {
+    stringified = [
+      'Query(',
+      Expr.toString(this.value),
+      ', ',
+      JSON.stringify(metadata),
+      ')',
+    ]
+  } else {
+    stringified = ['Query(', Expr.toString(this.value), ')']
+  }
+
+  return stringified.join('')
```

Comment:
```suggestion
  const notEmptyMetadata = Object.keys(metadata).some(key => metadata[key])

   if (Object.keys(metadata).some(key => metadata[key])) {
     return [
       'Query(',
       Expr.toString(this.value),
       ', ',
       JSON.stringify(metadata),
       ')',
     ].join('')
   }

   return ['Query(', Expr.toString(this.value), ')'].join('')
```

I usually don't like multiple `returns, but I think it would be fine here.

Anyway, the PR is good and feel free to discard this suggestion :)